File: code-coaches/easy/izzy-the-iguana/izzy_the_iguana.cpp
#include <iostream>
#include <unordered_map>
#include <vector>

int get_snack_points(const std::vector<std::string>& snacks)
{
    std::unordered_map<std::string, int> snacks_dict
    {
        {"Mango", 9},
        {"Lettuce", 5},
        {"Carrot", 4},
        {"Cheeseburger", 0},
    };

    int points = 0;
    for (const auto& snack : snacks)
    {
        points += snacks_dict[snack];
    }

    return points;
}

int main()
{
    std::vector<std::string> snacks;
    std::string snack;
    while (std::cin >> snack)
    {
        snacks.push_back(snack);
    }

    if (get_snack_points(snacks) >= 10)
        std::cout << "Come on Down!\n";
    else
        std::cout << "Time to wait\n";

    return 0;
}

File: code-coaches/easy/izzy-the-iguana/IzzyTheIguana.java
import java.util.*;

public class Main {
    public static int getSnackPoints(List<String> snacks) {
        Map<String, Integer> snacksDict = new HashMap<>();
        snacksDict.put("Mango", 9);
        snacksDict.put("Lettuce", 5);
        snacksDict.put("Carrot", 4);
        snacksDict.put("Cheeseburger", 0);

        int points = snacks.stream().mapToInt(snack -> snacksDict.getOrDefault(snack, 0)).sum();
        return points;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<String> snacks = Arrays.asList(scanner.nextLine().split("\\s+"));

        if (getSnackPoints(snacks) >= 10)
            System.out.println("Come on Down!");
        else
            System.out.println("Time to wait");

        scanner.close();
    }
}
